lib/hsa/mcwamp_hsa.cpp

```diff
@@ -5456,14 +5460,15 @@ HSACopy::enqueueAsyncCopy2dCommand(size_t width, size_t height, size_t srcPitch,
                 << "  copyAgent=" << copyDevice
                 << "\n");
         }
-        isAsync = true;
         
-        hcc_memory_async_copy_rect(getCommandKind(), copyDevice, dstPtrInfo, srcPtrInfo, width, height, srcPitch, dstPitch, depSignalCnt, depSignalCnt ? &depSignal:NULL, _signal);
+        status = hcc_memory_async_copy_rect(getCommandKind(), copyDevice, dstPtrInfo, srcPtrInfo, width, height, srcPitch, dstPitch, depSignalCnt, depSignalCnt ? &depSignal:NULL, _signal);
+        if(status != HSA_STATUS_SUCCESS)
+            return status;
```

Comment:
This path needs to release the signal (see line 5427) before returning

Reply:
I took another look again.  I think my previous comment about the failing path needs to release the signal is wrong since the signal release would be taken care of by the dtor of HSACopy.  Could you revert your last commit?